### PR TITLE
feat(conversation): show 1:1 conversation name (AR-1177)

### PR DESCRIPTION
**Pull request by vitorhugods**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1177" title="AR-1177" target="_blank">TaskAR-1177</a>  Display 1:1 conversation metadata
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Display name of the other user in the conversation list.

### Solutions

Use the new (not yet merged) `ObserveConversationListDetailsUseCase` from Kalium

### Dependencies

Needs releases with:

- [ ] https://github.com/wireapp/kalium/pull/350

#### How to Test

Just open the conversation list view

### Attachments (Optional)

Screenshot_1649113874

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-474/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3014789)(3014789)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3014789)(3014789)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-474/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3018668)(3018668)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3018668)(3018668)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-474/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3019338)(3019338)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3019338)(3019338)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt [hunk 1]
```diff
@@ -32,33 +37,27 @@ import javax.inject.Inject
...
+                state = ConversationListState(
+                    newActivities = listOf(),
+                    conversations = conversationMockData(detailedList.toGeneralConversationList()),
+                    missedCalls = mockMissedCalls,
+                    callHistory = mockCallHistory,
+                    unreadMentions = mockUnreadMentionList,
+                    allMentions = mockAllMentionList,
+                    unreadMentionsCount = 12,
+                    missedCallsCount = 100,
+                    newActivityCount = 1
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt:59]: Can we extract these values into constant values ?

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt [hunk 2]
```diff
@@ -126,31 +125,38 @@ class ConversationListViewModel @Inject constructor(
...
+            }
+            is OneOne -> {
+                val otherUser = details.otherUser
+                GeneralConversation(
+                    ConversationType.PrivateConversation(
+                        userInfo = UserInfo("", UserStatus.NONE), //TODO Get actual status and avatar
+                        conversationInfo = ConversationInfo(
+                            name = otherUser.name.orEmpty(),
+                            membership = Membership.None,
+                            isLegalHold = true
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt:150]: Legal hold is on by default ?

**Review by ohassine — APPROVED** [review]
lgtm 👌 I just left some tiny comments

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt (hunk 1 above)
Comment (vitorhugods) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt:59]: This is just mock data.
Needs to be removed.

**Review by trOnk12 — APPROVED** [review]
lgtm <3 awesome to see the UI being filled with data from Kalium <3

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-474/8/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt [hunk 3]
```diff
@@ -32,33 +37,27 @@ import javax.inject.Inject
...
+                    newActivities = listOf(),
+                    conversations = conversationMockData(detailedList.toGeneralConversationList()),
+                    missedCalls = mockMissedCalls,
+                    callHistory = mockCallHistory,
+                    unreadMentions = mockUnreadMentionList,
+                    allMentions = mockAllMentionList,
+                    unreadMentionsCount = 12,
+                    missedCallsCount = 100,
+                    newActivityCount = 1
+                )
```
[the remark this answers is not in the document]
Comment (vitorhugods) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationListViewModel.kt:60]: I'll do this in the next PR.

Today I just want to kill what's killing me, merge this and move on to the other conversation details.

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-474/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3020690)(3020690)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3020690)(3020690)-dev-release.apk)

**Pull request merged**